lib/web_ui/dev/integration_tests_manager.dart

```diff
@@ -391,11 +500,95 @@ class SafariIntegrationArguments extends IntegrationArguments {
       '--$mode',
       '--browser-name=safari',
       '--local-engine=host_debug_unopt',
+      if (isCanvaskitBackend) '--web-renderer=canvaskit',
     ];
   }
 
-  String getCommandToRun(String testName, String mode) =>
-      'flutter ${getTestArguments(testName, mode).join(' ')}';
+  String getCommandToRun(String testName, String mode,
+      {bool isCanvaskitBackend = false}) {
+    final String arguments =
+        getTestArguments(testName, mode, isCanvaskitBackend: isCanvaskitBackend)
+            .join(' ');
+    return 'flutter $arguments';
+  }
+}
+
+/// Parses additional options that can be used when running integration tests.
+class IntegrationTestsArgumentParser {
+  static final IntegrationTestsArgumentParser _singletonInstance =
+      IntegrationTestsArgumentParser._();
+
+  /// The [IntegrationTestsArgumentParser] singleton.
+  static IntegrationTestsArgumentParser get instance => _singletonInstance;
+
+  IntegrationTestsArgumentParser._();
+
+  /// If target name is provided integration tests can run that one test
+  /// instead of running all the tests.
+  String testTarget;
+
+  /// The build mode to run the integration tests.
+  ///
+  /// If not specified, these tests will run using 'debug, profile, release'
+  /// modes on Chrome and will run using 'profile, release' on other browsers.
+  ///
+  /// In order to skip a test for one of the modes, add the test to the
+  /// `blockedTestsListsMapForModes` list for the relevant compile mode.
+  String buildMode;
+
+  /// Whether to use html or canvaskit backend.
+  ///
+  /// If not set html rendering backend will be used.
+  String renderingBackend;
+
+  void populateOptions(ArgParser argParser) {
+    argParser
+      ..addOption(
+        'target',
+        defaultsTo: '',
+        help: 'By default integration tests are run for all the tests under'
+            'flutter/e2etests/web directory. If a test name is specified, that '
+            'only that test will run. The test name will be the name of the '
+            'integration test (e2e test) file. For example: '
+            'text_editing_integration.dart or '
+            'profile_diagnostics_integration.dart',
+      )
+      ..addOption('build-mode',
+          defaultsTo: '',
+          help: 'Flutter supports three modes when building your app. This '
+              'option sets the build mode for the integration tests. '
+              'By default an integration test will sequentially run on '
+              'multiple modes. All three modes (debug, release, profile) are '
+              'used for Chrome. Only profile, release modes will be used for '
+              'other browsers. In other words, if a build mode is selected '
+              'tests will only be run using that mode. '
+              'See https://flutter.dev/docs/testing/build-modes for more '
+              'details on the build modes.')
+      ..addOption('rendering-backend',
```

Comment:
For consistency with other tools, let's call it `--web-renderer`

Reply:
done. thanks for the suggestion!